src/common/utils.h

```diff
@@ -152,6 +153,16 @@ inline int GetExecNumMatchColor() {
   return std::min(num_match_color, GetNumThreadPerGPU());
 }
 
+template<typename T, typename V>
+V ParallelAccumulate(const T* a, const size_t n, V start) {
+  V sum = start;
+#pragma omp parallel for reduction(+:sum)
+  for (size_t i = 0; i < n; ++i) {
+    sum += a[i];
+  }
+  return sum;
+}
+
```

Comment:
T is for the element type of the array, and V is for the type of the accumulated sum.